lib/support/string/latin_dictionary.rb

```diff
@@ -0,0 +1,38 @@
+# lib/support/latin_dictionary.rb
+#
+module Support
+  module String
+    module LatinDictionary
+      def dictionary
+        {
+          length: { count: 0 },
+          uppercase: characters_to_dictionary(('A'..'Z').to_a),
+          lowercase: characters_to_dictionary(('a'..'z').to_a),
+          number: characters_to_dictionary(('0'..'9').to_a),
+          special: characters_to_dictionary([' ', '!', '@', '#', '$', '%', '^', '&', '*', '(', ')', '_', '+', '-', '=', '[', ']', '{', '}', '|', '"', '/', '\\', '.', ',', '`', '<', '>', ':', ';', '?', '~', "'"]),
+          unknown: {}
+        }
+      end
+
+      def dict_for_type(type)
+        full_dictionary = dictionary
```

Comment:
why don't we (simply) use dictionary explicitly?

Reply:
cause it will compute 3 times the hash, maybe its an useless performance improvement 😅